### DISCUSSION TITLE
Terminal Preview 1.20.10303.0 - Can't paste into vi on WSL

### DISCUSSION THREAD

**Discussion by Rigido71** [microsoft/terminal#16713] [Q&A] [answered]
Hi,
I'm pretty sure all started with latest upgrade: if I try to paste some text containing one or more . (point), no matter how (right mouse button, ctrl-V or shift-insert), into an Ubuntu WSL window ssh'ed to an AIX system, points are somehow interpreted as newline + cursor up. 
If you try to paste the following ip address 141.86.241.148 you'll have:

    +1  n148
    +2  n241
    +3  n86
    +4  141

(This happens on vi and sometimes on command line too.)

If you paste the previous ip on the Ubuntu WSL window itself you'll get "141n86n241n148" and if you paste it on vi on the ubuntu window you'll get "141^[On86^[On241^[On148".

Do someone have any hints on this?

**Comment by j4james** [comment, marked as answer]
I'm curious what keyboard layout you're using, and whether it makes any difference if you switch to something like _English United States_ before pasting. Because this looks like it might be a side effect of issue #16654.

What I think is happening is the paste operation is generating keypresses to represent the content on the clipboard, and for the `.` character it has chosen to use the `.` key on the numeric keypad (this is the part which I suspect may be dependent on your keyboard layout, because I can't reproduce the issue myself).

And the reason I think that fails, is because some editors enable [Keypad Application Mode](https://vt100.net/docs/vt510-rm/DECKPAM.html), which makes the terminal generate special escape sequence for numeric keypad keys. In particular, the `.` key on the keypad is expected to generate the sequence `^[On`.

We only added support for this mode in the last update, which is why you wouldn't have had noticed the issue before. It should be fixed in the next release though.

**Comment by Rigido71** [comment]
Hi j4james,
I can confirm that the layout IS the problem.
I'm on a Lenovo ThinkPad with italian keyboard layout, if I start TP with ITA layout the problem is there, no matter if I change the layout to US. But if I start TP with US layout the problem won't appear even if I switch to ITA without restarting app.
(I copied the IP once and pasted it many times.)

Thank you.
